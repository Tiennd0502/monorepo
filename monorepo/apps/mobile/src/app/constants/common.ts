export const CONGRATS_IMAGE = 'https://i.ibb.co/nkf6hFr/congrats.png';
