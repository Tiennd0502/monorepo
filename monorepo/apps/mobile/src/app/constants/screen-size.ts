import { Dimensions } from 'react-native';

export const BASE_SCREEN_WIDTH = 412;
export const BASE_SCREEN_HEIGHT = 842;

export const SCREEN_WIDTH: number = Dimensions.get('window').width;
export const SCREEN_HEIGHT: number = Dimensions.get('window').height;

export const CONTENT_WIDTH: number = SCREEN_WIDTH - 40;
