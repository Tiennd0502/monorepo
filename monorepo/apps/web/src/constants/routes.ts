export const ROUTES = {
  HOME: '/',
  VERIFY_OTP: '/verify-otp',
  SIGN_UP: '/sign-up',
  LOGIN: '/login',
};
