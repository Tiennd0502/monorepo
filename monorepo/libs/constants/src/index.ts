export * from './apis';
export * from './message';
export * from './commons';
export * from './product';
export * from './category';
export * from './validation';
