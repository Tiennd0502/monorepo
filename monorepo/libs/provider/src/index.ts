export { default as TamaguiProvider } from './TamaguiProvider';
export { default as QueryClientProvider } from './QueryClientProvider';
