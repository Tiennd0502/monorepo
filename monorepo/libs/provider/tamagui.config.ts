import { createTamagui, createTokens } from 'tamagui';
import { config } from '@tamagui/config/v3';
import { themes } from '@tamagui/themes';

import {
  colorsMapping,
  radius,
  size,
  space,
  zIndex,
  defaultFont,
  headingFont,
} from '@monorepo/ui';

const tokens = {
  radius,
  size,
  space,
  zIndex,
};

const tamaguiConfig = createTamagui({
  ...config,
  defaultFont: 'body',
  themes,
  fonts: {
    body: defaultFont,
    heading: headingFont,
  },
  tokens: createTokens({
    ...tokens,
    color: colorsMapping,
  }),
});

type AppConfig = typeof config;

declare module 'tamagui' {
  type TamaguiCustomConfig = AppConfig;
}

export default tamaguiConfig;
