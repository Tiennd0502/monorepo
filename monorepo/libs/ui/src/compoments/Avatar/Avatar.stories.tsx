import { Meta, StoryObj } from '@storybook/react';

import { USER } from '@monorepo/constants';

import Avatar from '.';

const meta: Meta<typeof Avatar> = {
  title: 'Components/Avatar',
  component: Avatar,
};

export default meta;

type Story = StoryObj<typeof Avatar>;

export const Default: Story = {
  args: USER,
};
