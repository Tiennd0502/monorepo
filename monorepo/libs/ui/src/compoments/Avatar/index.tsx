import { memo } from 'react';
import { Image, Stack, XStack } from 'tamagui';

import { AVATAR_DEFAULT } from '@monorepo/constants';
import Text from '../Text';

export interface AvatarProps {
  name: string;
  email: string;
  avatar?: string;
}

const Avatar = ({ name, email, avatar }: AvatarProps) => {
  return (
    <XStack gap="$5" alignItems="center">
      <Stack width="$20" height="$20" borderRadius="$20" overflow="hidden">
        <Image
          width="100%"
          height="100%"
          source={{ uri: avatar || AVATAR_DEFAULT }}
          resizeMode="cover"
          alt="Avatar"
        />
      </Stack>
      <Stack gap="$1">
        <Text color="$primary" fontSize="$5" fontWeight="bold">
          {name}
        </Text>
        <Text color="$textLabel">{email}</Text>
      </Stack>
    </XStack>
  );
};

export default memo(Avatar);
