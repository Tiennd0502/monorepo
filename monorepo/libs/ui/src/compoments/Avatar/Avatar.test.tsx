import {render} from '@testing-library/react-native';

import { TamaguiProvider } from '@monorepo/provider';
import {USER, AVATAR_DEFAULT} from '@monorepo/constants';

import Avatar from '.';

const baseProps = {
  name: USER.name,
  email: USER.email,
};

const setup = (moreProps = {}) => {
  const props = {
    ...baseProps,
    ...moreProps,
  };
  
  return render(
    <TamaguiProvider>
      <Avatar {...props} />
    </TamaguiProvider>
  );
};

describe('Avatar', () => {
  it('should render Avatar component with empty image', () => {
    const container = setup();

    expect(container).toMatchSnapshot();
  });

  it('should render Avatar component has src props', () => {
    const container =setup({avatar:AVATAR_DEFAULT});

    expect(container).toMatchSnapshot();
  });
});
