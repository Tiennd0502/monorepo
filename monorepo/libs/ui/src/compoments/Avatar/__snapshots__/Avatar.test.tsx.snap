// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render Avatar component has src props 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "gap": 20,
    }
  }
>
  <View
    style={
      {
        "borderBottomLeftRadius": 80,
        "borderBottomRightRadius": 80,
        "borderTopLeftRadius": 80,
        "borderTopRightRadius": 80,
        "height": 80,
        "overflow": "hidden",
        "width": 80,
      }
    }
  >
    <Image
      alt="Avatar"
      resizeMode="cover"
      source={
        {
          "uri": "https://i.ibb.co/zJrdJfJ/Ellipse-30.png",
        }
      }
      style={
        {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      {
        "gap": 4,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#242424",
          "fontFamily": "NunitoSans, Helvetica, Arial, sans-serif",
          "fontSize": 20,
          "fontWeight": "bold",
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      Bruno Pham
    </Text>
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#808080",
          "fontFamily": "NunitoSans_7pt-Medium",
          "fontSize": 14,
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      bruno203@gmail.com
    </Text>
  </View>
</View>
`;

exports[`Avatar should render Avatar component with empty image 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "gap": 20,
    }
  }
>
  <View
    style={
      {
        "borderBottomLeftRadius": 80,
        "borderBottomRightRadius": 80,
        "borderTopLeftRadius": 80,
        "borderTopRightRadius": 80,
        "height": 80,
        "overflow": "hidden",
        "width": 80,
      }
    }
  >
    <Image
      alt="Avatar"
      resizeMode="cover"
      source={
        {
          "uri": "https://i.ibb.co/zJrdJfJ/Ellipse-30.png",
        }
      }
      style={
        {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      {
        "gap": 4,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#242424",
          "fontFamily": "NunitoSans, Helvetica, Arial, sans-serif",
          "fontSize": 20,
          "fontWeight": "bold",
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      Bruno Pham
    </Text>
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#808080",
          "fontFamily": "NunitoSans_7pt-Medium",
          "fontSize": 14,
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      bruno203@gmail.com
    </Text>
  </View>
</View>
`;
